include/deal.II/lac/eigen.h

```diff
@@ -385,13 +387,17 @@ EigenInverse<VectorType>::solve (double           &value,
       // Compute unshifted eigenvalue
       value = (entry * x (i) < 0.) ? -length : length;
       value = 1./value;
-      value -= A_s.shift ();
+      value -= current_shift;
```

Comment:
This must be the most complicated way to write this expression. Would you mind taking the opportunity to rewrite this as
```
  value = (entry * x (i) < 0.) ? -1 : +1)/length - current_shift();
```
or similar?